SConstruct

```diff
@@ -1281,6 +1281,11 @@ if doConfigure :
 	if not c.CheckLibWithHeader( "blosc" + env["BLOSC_LIB_SUFFIX"], "blosc.h", "C++") :
 		sys.stderr.write( "ERROR : unable to find the Blosc libraries - check BLOSC_INCLUDE_PATH and BLOSC_LIB_PATH.\n" )
 		Exit( 1 )
+	
+	if c.CheckLibWithHeader( "freetype", ["ft2build.h"], "CXX" ) :
```

Comment:
Moving the freetype check here has the effect of adding `freetype` to the `LIBS` for `env`, causing _everything_ to link against freetype. Then intention was that only the libraries with a direct dependency would link it - for instance, IECoreImage does this :

```
if "-DIECORE_WITH_FREETYPE" in imageEnv["CPPFLAGS"] :
	imageEnv.Append( LIBS = "freetype" )
else :
	imageSources.remove( "src/IECoreImage/Font.cpp" )
	imagePythonSources.remove( "src/IECoreImageBindings/FontBinding.cpp" )
```

Can we put the call to `c.CheckLibWithHeader( "freetype", ["ft2build.h"], "CXX" )` back in its own `freetypeEnv` scope to avoid this "leak"? I think we can still put `-DIECORE_WITH_FREETYPE` in `env` as you have done here.